packages/core/src/loop/workflows/agentic-execution/llm-execution-step.ts

```diff
@@ -651,13 +651,20 @@ export function createLLMExecutionStep<
 
       const steps = inputData.output?.steps || [];
 
+      // Only include content from this iteration, not all accumulated content
+      // Get the number of existing response messages to know where this iteration starts
+      const existingResponseCount = inputData.messages?.nonUser?.length || 0;
+      const allResponseContent = messageList.get.response.aiV5.modelContent();
+
+      // Extract only the content added in this iteration
+      const currentIterationContent = allResponseContent.slice(existingResponseCount);
```

Comment:
I think the right fix might be to pass the step number into `stepContent(number)` to only return content for a specific step, and use `stepContent` instead of `modelContent` here. I meant to add that to `stepContent()` but didn't get around to it